File: backend/eswap/src/main/java/com/eswap/model/UserBalance.java
package com.eswap.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

@Entity
@Table(name = "user_balance")
@Getter
@Setter
public class UserBalance {
    @Id
    private Long userId;
    @Column(nullable = false)
    private BigDecimal balance = BigDecimal.ZERO;
    private String bankName;
    private String bankAccountNumber;
    private String accountHolder;
    private boolean isWithdrawRequested;
    private OffsetDateTime withdrawDateTime;
}

File: mobile/eswap/lib/model/user_balance.dart
class UserBalance {
  final int userId;
  double balance;
  final String? bankName;
  final String? bankAccountNumber;
  final String? accountHolder;
  bool withdrawRequested;
  DateTime? withdrawDateTime;

  UserBalance({
    required this.userId,
    required this.balance,
    this.bankName,
    this.bankAccountNumber,
    this.accountHolder,
    required this.withdrawRequested,
    this.withdrawDateTime,
  });

  factory UserBalance.fromJson(Map<String, dynamic> json) {
    return UserBalance(
      userId: json['userId'],
      balance: json['balance']?.toDouble() ?? 0.0,
      bankName: json['bankName'],
      bankAccountNumber: json['bankAccountNumber'],
      accountHolder: json['accountHolder'],
      withdrawRequested: json['withdrawRequested'] ?? false,
      withdrawDateTime: json['withdrawDateTime'] != null
          ? DateTime.parse(json['withdrawDateTime']).toLocal()
          : null,
    );
  }
}
